components/payment-term.jsx

```diff
@@ -143,31 +202,66 @@ export function PaymentTerm ({
 							{/* <br />Save up to 25% when you pay annually */}
 						</div>
 					) : (
-						<div>
-							<span
-								className={largePrice ? 'ncf__payment-term__large-price' : ''}
-							>
-								{option.price}
-							</span>
-							{option.chargeOnText && (
-								<p className="ncf__payment-term__charge-on-text">
-									{option.chargeOnText}
-								</p>
-							)}
-						</div>
+						// this should cover the cases different than annual, quarterly and monthly
+						// for those containing period on option.value, render custom template, for the rest keep legacy render
+						typeof option.value === 'string' && !!(option.value.match(/^(P[0-9]+(M|Y))$/)) ?
```

Comment:
more readable, less error prone
```suggestion
						typeof option.value === 'string' && Boolean(option.value.match(/^(P[0-9]+(M|Y))$/)) ?
```

Reply:
https://github.com/Financial-Times/n-conversion-forms/pull/632#discussion_r870693249